### DISCUSSION TITLE
能走多久和多远？

### DISCUSSION THREAD

**Discussion by xmsz-stu** [DouyinFE/semi-design#1895] [General]
目前在用semi

我们之前市面上几乎(ant/element/shoelace/arco...)都用过了，上一个用得最多的是fusion，而semi是现在想要主用的，因为

1. 主题和文档确实相对顺眼一点，可以看出至少有设计的水平，其他UI库说实话是完全没有设计师，纯粹只是前端UI封装库，所以之前我们用的主题都进行了大量修改，比如各种大小间距。而semi目前来说相对改的很少

2. 有很多新方向在探索，比如类似web component的设计很不错，D2C也很不错（虽然和之前我们用过很多类似产品的效果一样基本还是应用不上）

还有其他的，比如组件和属性的完整度，可能暂时还能接受，毕竟大部分场景下基础组件都够用了

然后，接下来是考虑稳定性的问题(对应是企业替换组件成本太高)，**semi什么情况下会停止更新?**。

比如常见的
- 主创走了/没激情了，团队其他人撑不起来或者团队其他人不想做了
- 业务没需求，项目被停止

....

还有一些疑问
- 为什么火山引擎改用arco了？使用的客户是不是变少了？
-

**Comment by pointhalo** [comment]
能理解对 lib 持续维护性的担忧，同时也能理解一些例如宣称永不弃坑的口号会显得很空洞。

作为这个项目的主要维护者，我可以适当给你一些数据以及我们的演进历史，希望能够给到你更多的上下文输入。

- 首先 semi design 它从本质上，属于是一个商业体（字节-抖音）里的技术团队（DouyinFE）的开源技术产品，是团队组织的产物，并不是一个个人或团队为爱发电的项目。它属于是内部技术项目逐渐成熟，能力溢出后选择开放的Case

- 我们的立项是在 2019.02，第一次  release 是2019.05，整体品牌设计风格以及 Token 方案，稳定沿用至今（实际上在当时的时间节点上，相比其他可选项，还是比较惊艳的，这也是我们在内部能迅速铺开，扩大落地规模的重要影响因子）。在内部持续打磨两年多，21年选择对外开源。期间一直保持着稳定的每双周迭代一个 minor 版本的频次，这些在我们的changelog页面有着完整的记录，至今其实已经稳定维持了接近 4 年半的时间了
![image](https://github.com/DouyinFE/semi-design/assets/88709023/efe51bcf-1e7a-4513-9274-97f247b52adf)

- 我们在抖音内部是数千个前端项目的依赖，不管外部开源社区这块的需求如何，公司内部就已经有着源源不断的业务更新维护需求会提过来。不管服务对象有没有继续扩大，我们投入的基础人力都是一个刚需（事实上，我们之前有做过统计， 自开源后，我们仍然有高达 40%的 Issue 来源于我们内部需求），选择开源对我们来说更像是一个水到渠成的选择，既然已经做了，就顺带开放给更大范围用户。同时我们自己有做一些自认为比较有亮点的事情（eg：Foundation/Adapter、DesignToken、DSM、D2C），想跟大家分享一下，过往的技术选型，实现原理参考等等也有受益于社区，想有更多的交流，更多的 insprire

- 如果有关注其他的一些 issue我们的回复（例如希望我们提供更多框架版本组件库的、希望我们除了figma 之外扩大 D2C 适用的设计平台，希望我们做各类 pro模版……），我们对于开新坑这个事情，其实是比较克制的。鉴于我们团队自身其实很轻量，只有少数几个人，只有我们判断一个方向是内外部同时存在强需求时，我们才会愿意去做比较大的投入。也只有保证对内也有价值，我们才能承诺持续投入。否则如果纯纯做一个仅服务于外部的产品同时不商业化，我们很快就会面临断粮的现实问题😂

- 不盲目开新坑，开了的就会稳定维护，这是我们对自身团队的要求。如果说我们会一直维护下去，不设期限，即使这是一个组织项目，我觉得也不太现实。但起码在可预见的中长期，内部需求不断，我们的人力投入就不会断。

同时外部贡献者也在逐渐增加喇，欢迎更多的 developer 一起贡献 ❤️ 🍻
